### PR TITLE
MicroProfile Platform 4.0

### PR DESCRIPTION
**Pull request by kabir**
https://issues.redhat.com/browse/WFLY-14277 - MP JWT 1.2 upgrade
https://issues.redhat.com/browse/WFLY-14261 - MP Health 3.0
https://issues.redhat.com/browse/WFLY-14230 - MP OpenTracing 2.0
https://issues.redhat.com/browse/WFLY-14201 - MP Config 2.0
https://issues.redhat.com/browse/WFLY-14245 - MP OpenAPI 2.0
https://issues.redhat.com/browse/WFLY-14281 - MP Fault Tolerance 3.0
https://issues.redhat.com/browse/WFLY-14206 - MP Metrics 3.0
https://issues.redhat.com/browse/WFLY-14312 - MP Rest Client 2.0

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -404,7 +406,7 @@
         <version.org.jboss.mod_cluster>1.4.1.Final</version.org.jboss.mod_cluster>
         <version.org.jboss.narayana>5.10.6.Final</version.org.jboss.narayana>
         <version.org.jboss.openjdk-orb>8.1.4.Final</version.org.jboss.openjdk-orb>
-        <version.org.jboss.resteasy>3.14.0.Final</version.org.jboss.resteasy>
+        <version.org.jboss.resteasy>3.15.0.Alpha1</version.org.jboss.resteasy>
```
Comment (kabir) [inline pom.xml:409]: @asoldano @ronsigal Is there a 3.15.0.Final yet?

Comment (ronsigal) [reply pom.xml:409]: Hey @kabir, that's exactly what I'm working on. As soon as I (or someone - Jason Lee offered to look at it) can figure out that OpenTracing test failure, I'm ready to go.

File: pom.xml [hunk 2]
```diff
@@ -300,24 +300,26 @@
...
-        <version.io.smallrye.open-api>2.0.16</version.io.smallrye.open-api>
-        <version.io.smallrye.opentracing>1.3.4</version.io.smallrye.opentracing>
+        <version.io.smallrye.smallrye-common>1.5.0</version.io.smallrye.smallrye-common>
+        <version.io.smallrye.smallrye-config>2.0.2</version.io.smallrye.smallrye-config>
+        <version.io.smallrye.smallrye-fault-tolerance>5.0.0</version.io.smallrye.smallrye-fault-tolerance>
+        <version.io.smallrye.smallrye-health>3.0.0</version.io.smallrye.smallrye-health>
+        <version.io.smallrye.smallrye-jwt>3.0.0</version.io.smallrye.smallrye-jwt>
+        <version.io.smallrye.smallrye-metrics>3.0.1</version.io.smallrye.smallrye-metrics>
+        <version.io.smallrye.open-api>2.1.1</version.io.smallrye.open-api>
+        <version.io.smallrye.opentracing>2.0.0-RC1</version.io.smallrye.opentracing>
```
Comment (kabir) [inline pom.xml:321]: @jasondlee We need a proper release of this do you know if one exists?

File: pom.xml [hunk 3]
```diff
@@ -354,20 +356,20 @@
...
-        <version.org.eclipse.microprofile.health.api>2.2</version.org.eclipse.microprofile.health.api>
-        <version.org.eclipse.microprofile.jwt.api>1.1.1</version.org.eclipse.microprofile.jwt.api>
-        <version.org.eclipse.microprofile.metrics.api>2.3</version.org.eclipse.microprofile.metrics.api>
-        <version.org.eclipse.microprofile.openapi.api>1.1.2</version.org.eclipse.microprofile.openapi.api>
-        <version.org.eclipse.microprofile.opentracing>1.3.3</version.org.eclipse.microprofile.opentracing>
+        <version.org.eclipse.microprofile.health.api>3.0</version.org.eclipse.microprofile.health.api>
+        <version.org.eclipse.microprofile.jwt.api>1.2</version.org.eclipse.microprofile.jwt.api>
+        <version.org.eclipse.microprofile.metrics.api>3.0</version.org.eclipse.microprofile.metrics.api>
+        <version.org.eclipse.microprofile.openapi>2.0</version.org.eclipse.microprofile.openapi>
+        <version.org.eclipse.microprofile.opentracing>2.0-RC2</version.org.eclipse.microprofile.opentracing>
```
Comment (kabir) [inline pom.xml:372]: @jasondlee We need a proper release of this

File: pom.xml [hunk 4]
```diff
@@ -300,24 +300,26 @@
...
-        <version.io.smallrye.opentracing>1.3.4</version.io.smallrye.opentracing>
+        <version.io.smallrye.smallrye-common>1.5.0</version.io.smallrye.smallrye-common>
+        <version.io.smallrye.smallrye-config>2.0.2</version.io.smallrye.smallrye-config>
+        <version.io.smallrye.smallrye-fault-tolerance>5.0.0</version.io.smallrye.smallrye-fault-tolerance>
+        <version.io.smallrye.smallrye-health>3.0.0</version.io.smallrye.smallrye-health>
+        <version.io.smallrye.smallrye-jwt>3.0.0</version.io.smallrye.smallrye-jwt>
+        <version.io.smallrye.smallrye-metrics>3.0.1</version.io.smallrye.smallrye-metrics>
+        <version.io.smallrye.open-api>2.1.1</version.io.smallrye.open-api>
+        <version.io.smallrye.opentracing>2.0.0-RC1</version.io.smallrye.opentracing>
+        <version.io.smallrye.smallrye-reactive-converters>2.0.0.RC3</version.io.smallrye.smallrye-reactive-converters>
```
Comment (kabir) [inline pom.xml:322]: @jasondlee / @rhusar this looks to have come in as part of the FP work. Is there a final release?

Comment (rhusar) [reply pom.xml:322]: @kabir Final was released 2 weeks ago - opening a PR now.

File: pom.xml (hunk 3 above)
Comment (jasondlee) [reply pom.xml:372]: PR pending...

File: pom.xml (hunk 4 above)
Comment (rhusar) [reply pom.xml:322]: @kabir PR is up https://github.com/wildfly/wildfly/pull/14003 - thanks for noticing this!

File: pom.xml (hunk 2 above)
Comment (jasondlee) [reply pom.xml:321]: https://github.com/wildfly/wildfly/pull/14004

File: pom.xml (hunk 3 above)
Comment (jasondlee) [reply pom.xml:372]: https://github.com/wildfly/wildfly/pull/14004

**Comment by rhusar** [comment]
🎉

**Pull request merged**